packages/react-scripts/config/paths.js

```diff
@@ -99,8 +101,10 @@ module.exports = {
   dotenv: resolveApp('.env'),
   appPath: resolveApp('.'),
   appBuild: resolveApp('build'),
-  appPublic: resolveApp('public'),
-  appHtml: resolveApp('public/index.html'),
+  appPublic: resolveApp(`public/${getServedPath(resolveApp('package.json'))}`),
+  appHtml: resolveOwn(
+    `public/${getServedPath(resolveApp('package.json'))}/index.html`
```

Comment:
Why do we need this? My intention is that HTML and other static resources are always directly in `public` folder but **are served** from a relative subpath when one exists.